File: pomodoro_app/lib/models/pomodoro.dart
class Pomodoro {

  final String id;
  final String userId;
  final String? taskId;
  final int duration;
  final bool completed;
  final DateTime createdAt;

  Pomodoro({
    required this.id,
    required this.userId,
    this.taskId,
    required this.duration,
    required this.completed,
    required this.createdAt,
  });

  factory Pomodoro.fromJson(Map<String, dynamic> json) {
    return Pomodoro(
      id: json["_id"],
      userId: json["userId"],
      taskId: json["taskId"],
      duration: json["duration"] ?? 0,
      completed: json["completed"] ?? false,
      createdAt: DateTime.parse(json["createdAt"]).toLocal(),
    );
  }

}

File: backend/models/Pomodoro.js
const mongoose = require("mongoose");

const pomodoroSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User"
  },
  taskId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Task",
    default: null
  },
  duration: Number, // phút
  completed: Boolean,
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model("Pomodoro", pomodoroSchema);